File: 2630-alternating-digit-sum/alternating-digit-sum.py
class Solution(object):
    def alternateDigitSum(self, n):
        b=0 
        s=0 
        if((len(str(n))-1)%2==0):
            k=2
        else:
            k=3
        while n>0:
            b=n%10
            n=n//10
            if(k%2==0):
                s=s+b
            else:
                s=s-b
            k +=1
        return(s)

File: 2630-alternating-digit-sum/alternating-digit-sum.java
class Solution {
    public int alternateDigitSum(int n) {
        int m,t=0,s=0;
        if(String.valueOf(n).length()%2==0){
            m=-1;
        }
        else{
            m=1;
        }
        while(n>0){
            t=n%10;
            n=n/10;
            s=s+m*t;
            m=m*-1;
        }
        return(s);
    }
}